### PR TITLE
Move config-getting to VectorBasePlatform

### PR DESCRIPTION
**Pull request by t3chguy**
+ in Electron get config via IPC from main process which has access to the "local" config.json override file and can make people happy :D
+ Remove bunch of duplicated code,
+ and move comments around to put them in the right place

image

Fixes https://github.com/vector-im/riot-web/issues/9476

Signed-off-by: Michael Telatynski <7t3chguy@gmail.com>

### CODE REVIEW

File: src/vector/getconfig.js [hunk 1]
```diff
@@ -32,7 +34,7 @@ export async function getVectorConfig(relativeLocation) {
     }
 }
 
-function getConfig(configJsonFilename) {
+export function getConfig(configJsonFilename) {
```
Comment (bwindels) [inline src/vector/getconfig.js]: couldn't spot any usage of this outside of this module?

Comment (t3chguy) [reply src/vector/getconfig.js]: There isn't, but this method was previously duplicated in index.js and there it was exported, I consolidated the two

Comment (bwindels) [reply src/vector/getconfig.js]: Makes it a bit clearer it's only used internally when making changing in the future ...

Comment (t3chguy) [reply src/vector/getconfig.js]: Sure, will do

**Pull request merged**